### PR TITLE
Navigation column should be resizable

### PR DESCRIPTION
**Pull request by raoyitao**
* The navigation column should be resizable by dragging it. The Batch
  or InteractiveReport components will have to own the navigation width
  state respectively, since the Assertion pane and Search box also need
  to know this size to be positioned correctly.

### CODE REVIEW

File: testplan/web_ui/testing/src/Nav/Column.js [hunk 1]
```diff
@@ -7,19 +7,70 @@ import {LIGHT_GREY} from "../Common/defaults";
...
+    this.setState(() => ({mouseHover: true}));
+  }
+
+  mouseLeave(e) {
+    this.setState(() => ({mouseHover: false}));
+  }
+
+  mouseDown(e) {
+    this.enableDragging = true;
+	document.addEventListener('mousemove', this.mouseMove, false);
```
Comment (ryan-collingham) [inline testplan/web_ui/testing/src/Nav/Column.js]: Some dodgy indentation going on here and in other places - please make sure to stick to 2-space indents. You'll need to set up your editor or IDE to set different tab widths in .js files and .py files.

File: testplan/web_ui/testing/src/Toolbar/FilterBox.js [hunk 2]
```diff
@@ -2,24 +2,28 @@ import React, {Component} from 'react';
...
-      style={{width: `${COLUMN_WIDTH}em`}}>
-      <input
-        className={css(styles.searchBoxInput)}
-        type="text"
-        placeholder="&#xf002; Filter"
-        onKeyUp={this.props.handleNavFilter} />
-    </div>
-  );
+  render() {
+    console.log('===Paint filterbox===');
```
Comment (ryan-collingham) [inline testplan/web_ui/testing/src/Toolbar/FilterBox.js]: Do we need to keep these temporary logs? I think they can be removed before this is merged.

File: testplan/web_ui/testing/src/Toolbar/Toolbar.js [hunk 3]
```diff
@@ -53,6 +53,7 @@ library.add(
  */
 class Toolbar extends Component {
   constructor(props) {
+    console.log('===init toolbar===');
```
Comment (ryan-collingham) [inline testplan/web_ui/testing/src/Toolbar/Toolbar.js]: Temporary log here can be removed.

File: testplan/web_ui/testing/src/Toolbar/Toolbar.js [hunk 4]
```diff
@@ -343,6 +347,7 @@ class Toolbar extends Component {
    * Render the toolbar component.
    */
   render() {
+    console.log('===Paint toolbar===');
```
Comment (ryan-collingham) [inline testplan/web_ui/testing/src/Toolbar/Toolbar.js]: and this one

File: testplan/web_ui/testing/src/Toolbar/FilterBox.js [hunk 5]
```diff
@@ -2,24 +2,28 @@ import React, {Component} from 'react';
 import PropTypes from 'prop-types';
 import {StyleSheet, css} from 'aphrodite';
 
-import {COLUMN_WIDTH} from "../Common/defaults";
+//import {COLUMN_WIDTH} from "../Common/defaults";
```
Comment (ryan-collingham) [inline testplan/web_ui/testing/src/Toolbar/FilterBox.js]: Please remove this commented-out code

File: testplan/web_ui/testing/src/Nav/Column.js [hunk 6]
```diff
@@ -7,19 +7,70 @@ import {LIGHT_GREY} from "../Common/defaults";
...
+    this.mouseEnter = this.mouseEnter.bind(this);
+    this.mouseLeave = this.mouseLeave.bind(this);
+    this.mouseDown = this.mouseDown.bind(this);
+    this.mouseMove = this.mouseMove.bind(this);
+    this.mouseUp = this.mouseUp.bind(this);
+
+    this.state = {
+      mouseHover: false,
+    };
+    this.enableDragging = false;
```
[the remark this answers is not in the document]
Comment (raoyitao) [reply testplan/web_ui/testing/src/Nav/Column.js]: I removed this enableDragging.

File: testplan/web_ui/testing/src/Nav/Column.js (hunk 1 above)
Comment (raoyitao) [reply testplan/web_ui/testing/src/Nav/Column.js]: carelessly type a "tab" and it looks like 4 space in pycharm, fix it .

File: testplan/web_ui/testing/src/Toolbar/FilterBox.js (hunk 2 above)
Comment (raoyitao) [reply testplan/web_ui/testing/src/Toolbar/FilterBox.js]: removed

File: testplan/web_ui/testing/src/Toolbar/Toolbar.js (hunk 3 above)
Comment (raoyitao) [reply testplan/web_ui/testing/src/Toolbar/Toolbar.js]: removed

File: testplan/web_ui/testing/src/Toolbar/Toolbar.js (hunk 4 above)
Comment (raoyitao) [reply testplan/web_ui/testing/src/Toolbar/Toolbar.js]: removed

File: testplan/web_ui/testing/src/Toolbar/FilterBox.js (hunk 5 above)
Comment (raoyitao) [reply testplan/web_ui/testing/src/Toolbar/FilterBox.js]: removed.

**Comment by raoyitao** [comment]
> Looks good overall, please just see my comments about managing state in the Column component and small tidy-ups.
> 
> To ensure the Travis build can pass, make sure to run `npm test` locally. You can update snapshots by hitting the `u` key at the test runner page, make sure to commit the snapshot changes and include them in this PR. Also make sure to run the linter with `npm run lint:fix`, this will automatically fix up some small errors (missing semi-colons etc.) for you so remember to commit these changes if any are made, for anything it can't fix automatically it should tell you what the problem is and you can fix yourself.

I also applied this feature to interactive UI, but there's a failure in test. In "interactiveReport.js", there is a piece of code:

const centerPane = GetCenterPane(
      this.state,
      this.props,
      reportFetchMessage,
      null,
      selectedEntries,
    );
seems I should remove this.props in this function then test can pass.

File: testplan/web_ui/testing/src/Report/InteractiveReport.js [hunk 7]
```diff
@@ -483,7 +483,7 @@ class InteractiveReport extends React.Component {
     );
     const centerPane = GetCenterPane(
       this.state,
-      this.props,
+      //this.props,
```
Comment (yuxuan-ms) [inline testplan/web_ui/testing/src/Report/InteractiveReport.js:486]: remove

File: testplan/web_ui/testing/src/Nav/Column.js [hunk 8]
```diff
@@ -7,19 +7,64 @@ import {LIGHT_GREY} from "../Common/defaults";
...
+    this.mouseDown = this.mouseDown.bind(this);
+    this.mouseMove = this.mouseMove.bind(this);
+    this.mouseUp = this.mouseUp.bind(this);
+
+    this.state = {
+      mouseHover: false,
+    };
+  }
+
+  mouseEnter(e) {
```
Comment (yuxuan-ms) [inline testplan/web_ui/testing/src/Nav/Column.js:24]: Why do you listen mouseenter event?

File: testplan/web_ui/testing/src/Nav/Column.js [hunk 9]
```diff
@@ -7,19 +7,64 @@ import {LIGHT_GREY} from "../Common/defaults";
...
+  }
+
+  mouseDown(e) {
+    document.addEventListener('mousemove', this.mouseMove, false);
+    document.addEventListener('mouseup', this.mouseUp, false);
+  }
+
+  mouseMove(e) {
+    let event = e || window.event ;
+    this.props.handleColumnResizing(`${event.clientX}px`);
```
Comment (yuxuan-ms) [inline testplan/web_ui/testing/src/Nav/Column.js:39]: event.stopPropagation()
event.preventDefault()

**Comment by ryanc414** [comment]
> > Looks good overall, please just see my comments about managing state in the Column component and small tidy-ups.
> > To ensure the Travis build can pass, make sure to run `npm test` locally. You can update snapshots by hitting the `u` key at the test runner page, make sure to commit the snapshot changes and include them in this PR. Also make sure to run the linter with `npm run lint:fix`, this will automatically fix up some small errors (missing semi-colons etc.) for you so remember to commit these changes if any are made, for anything it can't fix automatically it should tell you what the problem is and you can fix yourself.
> 
> I also applied this feature to interactive UI, but there's a failure in test. In "interactiveReport.js", there is a piece of code:
> 
> const centerPane = GetCenterPane(
> this.state,
> this.props,
> reportFetchMessage,
> null,
> selectedEntries,
> );
> seems I should remove this.props in this function then test can pass.

There is the same code in the BatchReport, so I don't think it is that casuing a difference. I don't think the function will work if you remove self.props from the function call.

File: testplan/web_ui/testing/src/Nav/Column.js [hunk 10]
```diff
@@ -7,19 +7,63 @@ import {LIGHT_GREY} from "../Common/defaults";
...
+
+  mouseDown(e) {
+    document.addEventListener('mousemove', this.mouseMove, false);
+    document.addEventListener('mouseup', this.mouseUp, false);
+  }
+
+  mouseMove(e) {
+    let event = e || window.event ;
+    event.stopPropagation();
+    if (event.clientX < 150) {
```
[the remark this answers is not in the document]
Comment (raoyitao) [reply testplan/web_ui/testing/src/Nav/Column.js:26]: moved to default.js

**Pull request merged**